src/compiler/types.ts

```diff
@@ -2860,9 +2873,22 @@ namespace ts {
         /* @internal */
         resolvedApparentType: Type;
         /* @internal */
+        resolvedIndexType: IndexType;
+        /* @internal */
+        resolvedIndexedAccessTypes: IndexedAccessType[];

```

Comment:
Yes, I will be sparse, in the same way that our `nodeLinks` and `symbolLinks` arrays are sparse. I believe this performs better than a map.